src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Helpers/BackupScheduleHelpers.ps1

```diff
@@ -0,0 +1,356 @@
+		Write-Debug -Message $DatasourceType
+		
+		# Log all the parameters
+		Write-Debug -Message "Policy: $Policy"
+		Write-Debug -Message "DatasourceType: $DatasourceType"
+		Write-Debug -Message "PolicySubType: $PolicySubType.Value"
+		Write-Debug -Message "HourlyInterval: $HourlyInterval.Value"
+		Write-Debug -Message "HourlyScheduleWindowDuration: $HourlyScheduleWindowDuration.Value"
+		Write-Debug -Message "EnableLogBackup: $EnableLogBackup.Value"
+		Write-Debug -Message "LogBackupFrequency: $LogBackupFrequency.Value"
+		Write-Debug -Message "EnableDifferentialBackup: $EnableDifferentialBackup.Value"
+		Write-Debug -Message "DifferentialRunDay: $DifferentialRunDay.Value"
+		Write-Debug -Message "DifferentialScheduleTime: $DifferentialScheduleTime.Value"
+		Write-Debug -Message "EnableIncrementalBackup: $EnableIncrementalBackup.Value"
+		Write-Debug -Message "IncrementalRunDay: $IncrementalRunDay.Value"
+		Write-Debug -Message "IncrementalScheduleTime: $IncrementalScheduleTime.Value"
+		Write-Debug -Message "BackupFrequency: $BackupFrequency.Value"
+		Write-Debug -Message "ScheduleRunDay: $ScheduleRunDay.Value"
+		Write-Debug -Message "ScheduleTime: $ScheduleTime.Value"
+		
+		$manifest = LoadManifest -DatasourceType $DatasourceType
+		
+		Write-Debug -Message "Manifest: $manifest" 
+		
+		$FullBackupPolicy = $Policy.SubProtectionPolicy | where { $_.PolicyType -match "Full" }
+		$LogBackupPolicy = $Policy.SubProtectionPolicy | where { $_.PolicyType -match "Log" }
+		$DifferentialPolicy = $Policy.SubProtectionPolicy | where { $_.PolicyType -match "Differential" }
+		$IncrementalPolicy = $Policy.SubProtectionPolicy | where { $_.PolicyType -match "Incremental" }
+		
+		# Validate BackupFrequency
+		
+		if(-not $BackupFrequency.Value) {
+			Write-Debug "BackupFrequency is empty"
+			if ($manifest.allowedSubProtectionPolicyTypes.Count -gt 1) {
+				# SAPHANA case				
+				if(-not $FullBackupPolicy.SchedulePolicy.ScheduleRunFrequency) {
+					$errormsg = "BackupFrequency cannot be empty for FullBackup"
+					throw $errormsg
+				}
+				else {
+					$BackupFrequency.Value = $FullBackupPolicy.SchedulePolicy.ScheduleRunFrequency
+				}
+			}
+			else {
+				# AzureVM case
+				Write-Debug "AzureVM case"
+				if(-not $Policy.SchedulePolicy.ScheduleRunFrequency) {
+					$errormsg = "BackupFrequency cannot be empty for FullBackup"
+					throw $errormsg
+				}
+				else {
+					Write-Debug "assigning BackupFrequency"
+					$BackupFrequency.Value = $Policy.SchedulePolicy.ScheduleRunFrequency
+				}
+			}
+		}
+		
+		Write-Debug -Message $BackupFrequency.Value
+		
+		if (-not $manifest.allowedBackupFrequencies.Contains($BackupFrequency.Value)) {
+			$allowedValues = [System.String]::Join(', ', $manifest.allowedBackupFrequencies)
+			$errormsg = "Specified BackupFrequency " + $BackupFrequency.Value + " is not supported for DatasourceType " + $DatasourceType + "`nAllowed values are: " + $allowedValues
+			throw $errormsg
+		}
+		
+		Write-Debug "Here 0"
```

Comment:
Please remove this debug message. There are similar messages below. Please remove them all.

Reply:
will do